src/pages/vaults/VaultDepositWithdrawForm.tsx

```diff
@@ -0,0 +1,637 @@
+      userBalanceUpdated !== userBalance,
+  });
+  const marginUsageDiff = renderDiffOutput({
+    type: OutputType.Percent,
+    value: marginUsage?.current,
+    newValue: marginUsageUpdated,
+    withDiff:
+      MustBigNumber(amount).gt(0) &&
+      marginUsageUpdated != null &&
+      marginUsage?.current !== marginUsageUpdated,
+  });
+
+  // todo i18n
+  const inputFormConfig =
+    selectedType === 'deposit'
+      ? {
+          formLabel: stringGetter({ key: STRING_KEYS.AMOUNT_TO_DEPOSIT }),
+          buttonLabel:
+            currentForm === 'confirm'
+              ? stringGetter({ key: STRING_KEYS.CONFIRM_DEPOSIT })
+              : stringGetter({ key: STRING_KEYS.PREVIEW_DEPOSIT }),
+          inputReceiptItems: [
+            {
+              key: 'cross-free-collateral',
+              label: stringGetter({ key: STRING_KEYS.CROSS_FREE_COLLATERAL }),
+              value: freeCollateralDiff,
+            },
+          ],
+          receiptItems: [
+            {
+              key: 'cross-margin-usage',
+              label: stringGetter({ key: STRING_KEYS.CROSS_MARGIN_USAGE }),
+              value: marginUsageDiff,
+            },
+            {
+              key: 'vault-balance',
+              label: stringGetter({ key: STRING_KEYS.YOUR_VAULT_BALANCE }),
+              value: vaultDiff,
+            },
+          ],
+          transactionTarget: {
+            label: stringGetter({ key: STRING_KEYS.PROTOCOL_VAULT }),
+            icon: 'vault' as const,
+          },
+        }
+      : {
+          formLabel: stringGetter({ key: STRING_KEYS.AMOUNT_TO_WITHDRAW }),
+          buttonLabel:
+            currentForm === 'confirm'
+              ? stringGetter({ key: STRING_KEYS.CONFIRM_WITHDRAW })
+              : stringGetter({ key: STRING_KEYS.PREVIEW_WITHDRAW }),
+          inputReceiptItems: [
+            {
+              key: 'vault-balance',
+              label: stringGetter({ key: STRING_KEYS.YOUR_VAULT_BALANCE }),
+              value: vaultDiff,
+            },
+          ],
+          receiptItems: [
+            {
+              key: 'cross-free-collateral',
+              label: stringGetter({ key: STRING_KEYS.CROSS_FREE_COLLATERAL }),
+              value: freeCollateralDiff,
+            },
+            {
+              key: 'slippage',
+              label: stringGetter({ key: STRING_KEYS.EST_SLIPPAGE }),
+              value: <Output type={OutputType.Percent} value={slippagePercent} />,
+            },
+            {
+              key: 'est amount',
+              label: stringGetter({ key: STRING_KEYS.EXPECTED_AMOUNT_RECEIVED }),
+              value: <Output type={OutputType.Fiat} value={estimatedWithdrawalAmount} />,
+            },
+          ],
+          transactionTarget: {
+            label: stringGetter({ key: STRING_KEYS.CROSS_ACCOUNT }),
+            icon: 'cross' as const,
+          },
+        };
+
+  const errorErrors = errors.filter((e) => e.type === AlertType.Error);
```

Comment:
[suggestion] can we think of a more descriptive name for this? even if it has to be weirdly long. the name `errorErrors` is just a bit confusing to me